src/hooks/useDragLeave.ts

```diff
@@ -0,0 +1,57 @@
+import { debounce } from 'lodash'
+import { useEffect, useRef } from 'react'
+import { useDispatch } from 'react-redux'
+import { updateHoveringPathActionCreator } from '../actions/updateHoveringPath'
+import useSortedContext from './useSortedContext'
+
+const DEBOUNCE_DELAY = 50
+let hoverCount = 0
+let debouncedSetHoveringPath: ReturnType<typeof debounce> | null = null
+
+/**
+ * Hook to capture the dragleave event and dispatch the dragInProgress action.
+ */
+const useDragLeave = ({ isDeepHovering }: { isDeepHovering: boolean }) => {
+  const dispatch = useDispatch()
+  const prevIsDeepHoveringRef = useRef(isDeepHovering)
+  const { hoveringOnDropEnd } = useSortedContext()
+
+  // Initialize the debounced function if it hasn't been already
+  if (!debouncedSetHoveringPath) {
+    debouncedSetHoveringPath = debounce(() => {
+      // Only set hoveringPath to undefined if hoverCount is still zero
+      dispatch(updateHoveringPathActionCreator({ path: undefined }))
+    }, DEBOUNCE_DELAY)
+  }
+
+  useEffect(() => {
+    if (isDeepHovering && !prevIsDeepHoveringRef.current) {
+      // Cursor has entered a drop target
+      hoverCount += 1
```

Comment:
I may be misunderstanding `hoverCount`, but should we not also be decrementing whenever leaving the drop target? Here, it looks like we only increment when entering and decrement once during the cleanup.

This _might_ be the reason I'm seeing this issue where the arrow sometimes fails to appear when leaving and reentering the drop target:

https://github.com/user-attachments/assets/2ebbd4ae-ab78-441e-bd98-7ce9635be3d2

Reply:
Hmm, well yes it is supposed to be decremented, I did have it in place though, not sure how it got removed. I'll add the decrementing of `hoverCount` in else-if statement. Thanks for pointing out.